frontend/packages/operator-lifecycle-manager/src/utils.ts

```diff
@@ -0,0 +1,33 @@
+import * as _ from 'lodash';
+import { ClusterServiceVersionKind, CRDDescription } from './types';
+import { referenceForProvidedAPI } from './components';
+
+export const getInternalObjects = (csv: ClusterServiceVersionKind) => {
+  const internals: string =
+    csv?.metadata?.annotations?.['operators.operatorframework.io/internal-objects'];
+  if (!internals) {
+    return [];
+  }
+  try {
+    return JSON.parse(internals);
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.error('Error parsing internal object annotation.', e);
+    return [];
+  }
+};
+
+export const isInternalObject = (internalObjects: string[], objectName: string): boolean =>
+  internalObjects.some((obj) => obj === objectName);
+
+export const getInternalAPIReferences = (csv: ClusterServiceVersionKind): string[] => {
+  const owned: CRDDescription[] = _.flatten(
```

Comment:
Isn't `spec.customresourcedefinitions.owned` an array? I'd expect it to be simply

```ts
onst owned: CRDDescription[] = csv?.spec?.customresourcedefinitions?.owned || [];
```

Reply:
Yes, you're right. I was doing this because I was taking all CRDs both owned and required. #shithappens ;)